src/analyses/variable-sensitivity/constant_pointer_abstract_object.cpp

```diff
@@ -72,6 +72,48 @@ abstract_object_pointert constant_pointer_abstract_objectt::merge(
   return abstract_pointer_objectt::merge(other, widen_mode);
 }
 
+bool constant_pointer_abstract_objectt::same_target(
+  abstract_object_pointert other) const
+{
+  auto cast_other =
+    std::dynamic_pointer_cast<const constant_pointer_abstract_objectt>(other);
+
+  if(value_stack.is_top_value() || cast_other->value_stack.is_top_value())
+    return false;
+
+  if(value_stack.depth() != cast_other->value_stack.depth())
+    return false;
+
+  for(size_t d = 0; d != value_stack.depth() - 1; ++d)
+    if(
+      value_stack.target_expression(d) !=
+      cast_other->value_stack.target_expression(d))
+      return false;
+
+  return true;
+}
+
+exprt constant_pointer_abstract_objectt::offset() const
+{
+  if(value_stack.is_top_value())
+    return nil_exprt();
+  return value_stack.offset_expression();
+}
+
+exprt constant_pointer_abstract_objectt::offset_from(
+  abstract_object_pointert other) const
+{
+  auto cast_other =
+    std::dynamic_pointer_cast<const constant_pointer_abstract_objectt>(other);
+
+  if(value_stack.is_top_value() || cast_other->value_stack.is_top_value())
+    return nil_exprt();
+
+  return minus_exprt(
+    value_stack.offset_expression(),
+    cast_other->value_stack.offset_expression());
```

Comment:
Why convert back to exprt's when you could just subtract the two abstract objects?

Reply:
I'm not :)